src/cloudant/design_document.py

```diff
@@ -79,6 +89,22 @@ def add_view(self, view_name, map_func, reduce_func=None, **kwargs):
         view = View(self, view_name, map_func, reduce_func, **kwargs)
         self.views.__setitem__(view_name, view)
 
+    def add_search_index(self, search_index_name, search_index, **kwargs):
+        """
+        Appends a Cloudant search index to the locally cached DesignDocument
+        SearchIndex dictionary.
+
+        :param str search_index_name: Name used to identify the SearchIndex.
+        :param str search_index: Javascript search index function.
+        """
+        if self.get_search_index(search_index_name) is not None:
+            msg = "Search index {0} already exists in this design doc"\
```

Comment:
Also, use single quotes not double quotes to match the code style in the rest of the library.

Again, applies throughout the PR.